### ISSUE TITLE
np.linalg.norm grad fails on zero vectors

### ISSUE THREAD

**Issue by mattjj** [HIPS/autograd#151]
This prints NaNs:

```python
import autograd.numpy as np
from autograd.numpy.linalg import norm
from autograd import grad

print grad(norm)(np.zeros(3))
```

**Comment by j-towns** [comment]
The standard norm is not differentiable at zero. `autograd.numpy.abs` seems ot return a zero gradient at zero even though it is not differentiable there. Is this the behaviour we'd want for the norms too?

**Comment by mattjj** [comment]
Good point :) I was thinking norm squared.

I think returning zero would be nicer behavior, though you're right that maybe it should blow up here. I used norm in an objective function and tested exact optimality, then I was surprised it gave nans, but didn't have time to think about it more. 

What do you think? Let it generate nans, or just return zero for potential convenience like abs?

**Comment by mattjj** [comment]
I briefly discussed with @duvenaud and we think it's probably a good idea for the code in the OP to produce zeroes (and not nans). In other cases when the derivative is undefined we sometimes choose a reasonable convention for it (e.g. choosing an element of the subgradient) instead of blowing up with nans, as you pointed out for `np.abs` and also for `np.max` in #141.

**Comment by j-towns** [comment]
I agree, will try to get round to fixing this at some point. Do you think it's unnecessary to issue a warning in this case?
